### PR TITLE
Save and plot baremetal benchmark results

### PR DESCRIPTION
**Pull request by frdeso**
-Change lava submission script to Python3
-Post IRC notification on test failure

### CODE REVIEW

File: scripts/lttng-baremetal-tests/summarize-results.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -ex
```
Comment (mjeanson) [inline scripts/lttng-baremetal-tests/summarize-results.sh]: Missing "-u" to fail on undefined variables.

Comment (frdeso) [inline scripts/lttng-baremetal-tests/summarize-results.sh]: Done

File: scripts/lttng-baremetal-tests/lava-submit.py [hunk 2]
```diff
@@ -66,6 +66,22 @@ def check_job_all_test_cases_state_count(server, job):
                     failed_tests+=1
     return (passed_tests, failed_tests)
 
+# Get the benchmark results from the lava bundle
+# save them as CSV files localy
+def fetch_benchmark_results(server, job):
+    content = get_job_bundle_content(server, job)
+    testcases = ['processed_results_close.csv', 'processed_results_open_enoent.csv', 'processed_results_open_efault.csv']
+
+    for run in content['test_runs']:
```
Comment (PSRCode) [inline scripts/lttng-baremetal-tests/lava-submit.py]: Any other way to do this ?

File: scripts/lttng-baremetal-tests/lava-submit.py [hunk 3]
```diff
@@ -328,6 +343,13 @@ def main():
         print('argument -t/--type {} unrecognized. Exiting...'.format(args.type))
         return -1
 
+    lava_api_key = None
+    try:
+        lava_api_key = os.environ['LAVA_FRDESO_TOKEN']
```
Comment (PSRCode) [inline scripts/lttng-baremetal-tests/lava-submit.py]: As said in global comment. We might want to use LAVA_JENKINS_TOKEN instead.

Comment (frdeso) [inline scripts/lttng-baremetal-tests/lava-submit.py]: I agree

File: scripts/lttng-baremetal-tests/lava-submit.py (hunk 2 above)
Comment (frdeso) [inline scripts/lttng-baremetal-tests/lava-submit.py]: The result bundle is a JSON containing the results for every testcases as well as the files that were attached during the run.  We need to iterate over this JSON(once converted to a Python Dictionnary) to save the benchmark result files.
I could rename the variables to make it clearer what each step is doing. Would that work?

Comment (PSRCode) [inline scripts/lttng-baremetal-tests/lava-submit.py]: Variables are not that bad. Might want to add your reply as a comment and *maybe include a skeleton of the json structure you are extracting from.

File: scripts/lttng-baremetal-tests/lava-submit.py [hunk 4]
```diff
@@ -150,7 +150,8 @@ def get_boot_cmd():
 def get_config_cmd(build_device):
     packages=['bsdtar', 'psmisc', 'wget', 'python3', 'python3-pip', \
             'libglib2.0-dev', 'libffi-dev', 'elfutils', 'libdw-dev', \
-            'libelf-dev', 'libmount-dev', 'libxml2', 'libpfm4-dev']
+            'libelf-dev', 'libmount-dev', 'libxml2', 'libpfm4-dev', \
+            'libnuma-dev']
```
Comment (frdeso) [inline scripts/lttng-baremetal-tests/lava-submit.py]: I added those lines to the PR.

File: scripts/lttng-baremetal-tests/lava-submit.py (hunk 2 above)
Comment (frdeso) [inline scripts/lttng-baremetal-tests/lava-submit.py]: Done

**Pull request merged**